core/lib/spree/app_configuration.rb

```diff
@@ -173,6 +173,10 @@ class AppConfiguration < Preferences::Configuration
     #   @return [String] Email address used as +From:+ field in transactional emails.
     preference :mails_from, :string, default: 'spree@example.com'
 
+    # @!attribute [rw] manually_associate_user
+    #   @return [Boolean] manually call associate_user instead of relying on current_order
+    preference :manually_associate_user, :boolean, default: true
```

Comment:
Can you add something to the pref name that makes it clear that the association is responsibility of the authentication system? Like `manually_associate_user_in_authentication_extension`, or even without the initial `manually_`.

Reply:
Also the default (contrary to what I initially said 😅) should be `false`, to maintain the current behavior.